src/main/scala/cromwell/engine/backend/jes/authentication/JesAuthentication.scala

```diff
@@ -0,0 +1,66 @@
+package cromwell.engine.backend.jes.authentication
+
+import cromwell.engine.WorkflowDescriptor
+import cromwell.engine.backend.jes.{JesConfigurationSingleton, JesConfiguration, JesInterface}
+import spray.json.JsObject
+
+import scala.util.{Failure, Success}
+
+/**
+ * This makes JesConnection non visible when mixing a JesAuthentication trait
+ */
+trait JesConnection {
+  val JesConnection: JesInterface
+}
+
+/**
+ * Trait for JesAuthentication
+ */
+trait JesAuthentication { self: JesConfiguration with JesConnection =>
+
+  private def getRefreshToken(descriptor: WorkflowDescriptor) = descriptor.workflowOptions.get("refresh_token")
+  private def getAccessToken(descriptor: WorkflowDescriptor) = descriptor.workflowOptions.get("access_token")
+
+  def authenticated[A](f: JesInterface => A)(implicit descriptor: WorkflowDescriptor) = {
```

Comment:
I'm curious why you choose the implicit parameter considering that most usages of this function specify the value anyway.  What about something like this:

```scala
authenticated(descriptor) { connection =>
  ...
}
```

instead of 

```scala
authenticated { connection =>
  ...
} (descriptor)
```